build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,229 @@
+			description: "ADO Personal Access Token",
+			required: true,
+			env: "ADO_PAT",
+		}),
+		sourceBranch: Flags.string({
+			description: "Branch name across which the dev release manifest should be generated.",
+			required: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const flags = this.flags;
+
+		const PACKAGE_NAME = "@fluidframework/container-runtime";
+		const GITHUB_RELEASE_URL = `https://api.github.com/repos/microsoft/${flags.repoName}/releases`;
+		const ADO_BASE_URL = `https://dev.azure.com/${flags.repoName.toLowerCase()}/internal/_apis/build/builds?api-version=7.0`;
+		const REGISTRY_URL = `https://pkgs.dev.azure.com/${flags.repoName.toLowerCase()}/internal/_packaging/build/npm/registry/`;
+
+		// Authorization header for Azure DevOps
+		const authHeader = `Basic ${Buffer.from(`:${flags.ado_pat}`).toString("base64")}`;
+
+		if (authHeader === undefined || authHeader === null) {
+			this.error("Check your ADO Personal Access Token. It maybe incorrect or expired.");
+		}
+
+		try {
+			const buildNumber = await getFirstSuccessfulBuild(
+				authHeader,
+				ADO_BASE_URL,
+				flags.sourceBranch,
+				this.logger,
+			);
+			if (buildNumber !== undefined) {
+				this.log(
+					`Most successful build number for the last 24 hours for ${flags.sourceBranch} branch: ${buildNumber}`,
+				);
+				const devVersion = await fetchDevVersionNumber(
+					authHeader,
+					REGISTRY_URL,
+					PACKAGE_NAME,
+					buildNumber,
+					this.logger,
+				);
+				if (devVersion !== undefined) {
+					this.log(`Fetched dev version: ${devVersion}`);
+					const manifest = await generateManifestObjectForDevReleases(
+						GITHUB_RELEASE_URL,
+						this.logger,
+					);
+					if (manifest !== undefined) {
+						await writeManifestToFile(devVersion, manifest, this.logger);
+					}
+				}
+			} else if (buildNumber === undefined) {
+				this.log(
+					`No successful build found for ${flags.sourceBranch} branch in the last 24 hours`,
+				);
+			}
+		} catch (error: unknown) {
+			this.errorLog(`Error creating manifest file: ${error}`);
+		}
+	}
+}
+
+/**
+ * Fetches the first successful build number in the last 24 hours.
+ * @returns The build number if successful, otherwise undefined.
+ */
+async function getFirstSuccessfulBuild(
+	authHeader: string,
+	adoUrl: string,
+	sourceBranch: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	try {
+		const response = await fetch(adoUrl, { headers: { Authorization: authHeader } });
+		const data = await response.json();
+
+		const twentyFourHoursAgo = new Date();
+		twentyFourHoursAgo.setHours(twentyFourHoursAgo.getHours() - 24);
+
+		const successfulBuilds = data.value.filter(
+			(build: IBuildDetails) =>
+				build.definition.name === "Build - client packages" &&
+				build.status === "completed" &&
+				build.result === "succeeded" &&
+				build.sourceBranch === `refs/heads/${sourceBranch}` &&
+				new Date(build.finishTime) >= twentyFourHoursAgo,
```

Comment:
I would do the date check first. That way you'll only be running the rest of the checks on the number of builds run in a 24-hour period. As it is, you'll check the name of all builds from all time.